components/Makefile

```diff
@@ -98,7 +98,10 @@ clobber:	$(COMPONENT_DIRS.nosetup) clean
 	$(RM) -r $(WS_REPO) $(WS_LOGS) $(WS_LINT_CACHE)
 
 
-setup:	$(WS_LOGS) $(WS_REPO) tools $(WS_LINT_CACHE)
+setup:	install_build_essential $(WS_LOGS) $(WS_REPO) tools $(WS_LINT_CACHE)
+
+install_build_essential:
+	pfexec pkg install -v build-essential || [ $$? -eq 4 ] && true || false
```

Comment:
Please, borrow logic from make-rules/environment.mk - pkg list || pkg install - this is much faster when package is already installed.